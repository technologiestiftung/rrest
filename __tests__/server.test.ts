import request from 'supertest';
import app from '../node-app/src/lib/server';

describe('default testing for server', () => {
  it('should response with 200 on /', async () => {
    expect.assertions(2);
    const response = await request(app.callback()).get('/');
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });
  it('should accept post', async () => {
    expect.assertions(1);
    const response = await request(app.callback())
      .post('/submit')
      .send({foo: 'bah'});
    expect(response.status).toBe(200);
  });
  it('should return the passed object', async () => {
    expect.assertions(1);
    const obj = {foo: 'bah'};
    const response = await request(app.callback())
      .post('/submit')
      .send(obj);
    expect(response.text).toEqual('{"code":0,"data":{"foo":"bah"},"errors":""}');
  });
});
