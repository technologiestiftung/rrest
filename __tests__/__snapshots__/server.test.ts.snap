// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default testing for server should response with 200 on / 1`] = `
"<!doctype html>

<html lang=\\"en\\">

<head>
  <meta charset=\\"utf-8\\">
  <meta http-equiv=\\"x-ua-compatible\\" content=\\"ie=edge\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
  <!-- The above 3 meta tags *must* come first in the head; any other head content must come *after* these tags -->
  <title>Page Title</title>
  <meta name=\\"description\\" content=\\"something\\">
  <meta name=\\"author\\" content=\\"me\\">
</head>

<body>
  <div class=\\"main\\">
    <form action=\\"http://locahost:3000/submit\\" method=\\"get\\" class=\\"form\\">
      <div>
        <label for=\\"content\\">
          content as JSON
        </label>
        <input type=\\"text\\" name=\\"content\\" id=\\"content\\" required>
        <div>
          <input type=\\"submit\\" value=\\"Send! \\">
        </div>
      </div>
    </form>
  </div>
</body>

</html>"
`;
